src/lib.rs

```diff
@@ -81,6 +82,20 @@ pub struct Pool<C: ManageConnection + Send> {
     conn_pool: Arc<ConnectionPool<C>>,
 }
 
+impl<C: ManageConnection + Send + fmt::Debug> fmt::Debug for Pool<C> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let (total, idle) = futures::executor::block_on(async {
```

Comment:
This seems a big heavy for a Debug implementation - I think we should just omit the information if it's an asynchronous task (I'm assuming to get the future locks?)

Reply:
Yes, it does have to lock a mutex to get those fields. I removed them